src/sif/motorscootercost.cc

```diff
@@ -75,8 +75,6 @@ constexpr ranged_default_t<float> kCountryCrossingPenaltyRange{0, kDefaultCountr
 constexpr ranged_default_t<float> kUseFerryRange{0, kDefaultUseFerry, 1.0f};
 constexpr ranged_default_t<float> kUseHillsRange{0, kDefaultUseHills, 1.0f};
 constexpr ranged_default_t<float> kUsePrimaryRange{0, kDefaultUsePrimary, 1.0f};
-constexpr ranged_default_t<uint32_t> kTopSpeedRange{kMinimumTopSpeed, kDefaultTopSpeed,
-                                                    kMaximumTopSpeed};
```

Comment:
I'm not sure how to deal with this in c++.. Motorscooter needs `kDefaultTopSpeed` (45 kph) as fallback if the specified `top_speed` was outside the acceptable range.. Right now it'll default to `kMaxSpeedKph`.. Not sure how to make this available by default in `dynamiccost.h` so that motor scooter can override the range check..

Reply:
we have a bit of an unfortunate situation here where all of our options parsing is not polymorphic. parsing is tied to specific classes but there is no class inheritance at play. so the standard object oriented approach wont work where the base class has functionality overridden in the child class. what that means is we're kind of screwed here without a redesign.

there is a work around where, in the parsing function for the class that needs to override this, you can modify `top_speed_` to be in range. so it starts out being wrong when DynamicCosts generic parsing function sets it but then in the motroscooter parse function we "fix" it.

id say for now that is an ok way to go. ill give some more thought about it in the next day or so in case there is another more elegant way.

Reply:
finally had some time again for this. I restored most of the code for motorscooter, so it's overriding the defaults by dynamiccost. If you have a better way, feel free to to override:)